app/models/publications/national_recruitment_performance_report_generator.rb

```diff
@@ -0,0 +1,25 @@
+module Publications
+  class NationalRecruitmentPerformanceReportGenerator
+    attr_reader :client,
+                :generation_date,
+                :publication_date,
+                :report_expected_time,
+                :cycle_week
+
+    def initialize(cycle_week:, generation_date: Time.zone.now, publication_date: nil)
+      @cycle_week = cycle_week
+      @generation_date = generation_date
```

Comment:
I can see the publication date and generation date are the same for now. I'm worried they will always be the same and we'll just have these two identical fields to confuse people. I know I've said it before, but I would prefer we just have one until we are really clear what the difference is and what we are going to do with them. (I will shut up about it now. This is my last comment on the subject)